src/os/bluestore/BlueStore.cc

```diff
@@ -2196,7 +2196,8 @@ void BlueStore::ExtentMap::reshard(
     if (spanning_blob_map.empty()) {
       bid = 0;
     } else {
-      bid = spanning_blob_map.rbegin()->first + 1;
+      int last_bid = spanning_blob_map.rbegin()->first;
+      bid = (last_bid == SHRT_MAX)?0:(last_bid+1);
```

Comment:
Hmm, I think a better fix here is to use a helper allocate_spanning_blob_id() that will return an unused id, called from both locations.  Otherwise we still risk a collision after we wrap.  The method can return 0 if the map is empty, end + 1 if it is small enough, otherwise pick a random number, seek to that position, and iterate through the map looking for an unused value.